### PR TITLE
Add Omnibox component

### PR DESCRIPTION
**Pull request by progfay**
[Image from Gyazo](https://gyazo.com/15a640c0bf95279dbb5ca06af75aad06)

From #16 .

## Usage

```jsx
<MapView
	options={{
		omniboxControl: true,
		omniboxControlOptions: {
			types: ['(cities)'],
			componentRestrictions: {
				country: ['jp'],
			},
			onPlaceChanged: (place) => { console.log(place) },
			styles: {
				omnibox: {},
				textInput: {},
				searchButton: {},
				searchIcon: {},
			},
		},
	}}
/>
```

## Props

```ts
// PlaceResult: https://developers.google.com/maps/documentation/javascript/reference/places-service#PlaceResult
type OmniboxCustomizeOptions = { onPlaceChanged?: (place: PlaceResult) => void }

// ViewStyleProps: https://facebook.github.io/react-native/docs/view-style-props
// TextStyleProps: https://facebook.github.io/react-native/docs/0.7/text#style
// ImageStyleProps: https://facebook.github.io/react-native/docs/image-style-props
type CustomOmniboxStyles = {
	styles: {
		omnibox: ViewStyleProps
		textInput: TextStyleProps
		searchButton: ViewStyleProps
		searchIcon: ImageStyleProps
	}
}

// AutocomplateOptions: https://developers.google.com/maps/documentation/javascript/reference/places-widget#AutocompleteOptions
type OmniboxControlOptions = AutocompleteOptions & OmniboxCustomizeOptions & CustomOmniboxStyles

// MapView.props.options: Options
type Options =  {
	// Show `<Omnibox>` when `true`
	omniboxControl: boolean,
	omniboxControlOptions: OmniboxControlOptions,
}
```

### CODE REVIEW

File: src/Omnibox.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      lat: lat(),
+      lng: lng(),
+    });
+  }
+
+  render() {
+    if (!this.props.map || !this.props.maps) return null;
+
+    return (
+      <View style={styles.omnibox}>
```
Comment (progfay) [inline src/Omnibox.js]: [Screenshot from Gyazo](https://gyazo.com/d784d4ec1d6d5d516ed6ade24504ad17)

I don't know how to apply `height` property for `<input>` .

File: src/Omnibox.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+});
+
+class MapOmnibox extends Component {
+  constructor(props) {
+    super(props);
+    this.onPlaceChanged = this.onPlaceChanged.bind(this);
+  }
+
+  onPlaceChanged() {
```
[the remark this answers is not in the document]
Comment (kasajei) [reply src/Omnibox.js]: Yes! 

- Default behavior  is good 
- We can pass through this.props.onPlaceChanged from parents

File: src/Omnibox.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      lng: lng(),
+    });
+  }
+
+  render() {
+    if (!this.props.map || !this.props.maps) return null;
+
+    return (
+      <View style={styles.omnibox}>
+        <input
```
[the remark this answers is not in the document]
Comment (progfay) [reply src/Omnibox.js]: Access with `ref._node` .

File: src/Omnibox.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+          ref={ref => {
+            if (!ref) return;
+            this.autocomplete = new this.props.maps.places.Autocomplete(
+              ref._node,
+              this.props.options
+            );
+            this.autocomplete.addListener('place_changed', this.onPlaceChanged);
+          }}
+        />
+        <style> {'#omnibox:focus { outline: none }'} </style>
```
Comment (progfay) [inline src/Omnibox.js]: To remove blue border of be focused `<TextInput>` .

**Comment by progfay** [comment]
I think about making customizable style of Omnibox.

```js
<MapView
	options={{
		omniboxControl: true,
		omniboxControlOptions: {
			styles: {{
				textInput: {/* style for <TextInput> of Omnibox */},
				container: {/* style for container of <TextInput> */},
			}},
		},
	}}
/>
```

**Pull request merged**